app/components/avo/fields/common/single_file_viewer_component.rb

```diff
@@ -40,4 +40,12 @@ def is_video?
   rescue
     false
   end
+
+  # If model is not persistent blob is automatically destroyed otherwise it can be "lost" on storage
+  def persistent_record?
+    return true if @resource.model.persisted?
+
+    ActiveStorage::Blob.destroy(file.blob_id) if file.blob_id.present?
```

Comment:
Will this have no side-effects when updating an existing record?

Reply:
No. On update `@resource.model.persisted?` returns `true`

Reply:
Now it's `file.blob_id.present?` not `@resource.model.persisted?`, so I'm a bit confused. Is it ok like this?

![CleanShot 2022-08-29 at 17 13 17](https://user-images.githubusercontent.com/1334409/187221734-08cde81c-79d2-4007-8df9-2fc694801451.jpg)